client/components/SignIn.jsx

```diff
@@ -1,16 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
+
+export default function SignIn({ action }) {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ username, password })
+    };
+    fetch('/api/auth/sign-up', req)
+      .then((res) => {
+        res.json();
+      })
+      .then((result) => {
```

Comment:
I suggest you log the result until it gets implemented, to help you in debugging and testing.